test/e2e/kuma_test.go

```diff
@@ -0,0 +1,145 @@
+		},
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: 0,
+		},
+	}
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).UpdateScale(ctx,
+		deployment.Name, scale, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	scale.Spec.Replicas = 2
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).UpdateScale(ctx,
+		deployment.Name, scale, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	t.Log("running ingress tests to verify all-in-one deployed ingress controller and proxy are functional")
+	deployIngress(ctx, t, env)
+	service, err := env.Cluster().Client().CoreV1().Services("default").Get(ctx, "httpbin", metav1.GetOptions{})
+	service.ObjectMeta.Annotations["ingress.kubernetes.io/service-upstream"] = "true"
+	service, err = env.Cluster().Client().CoreV1().Services("default").Update(ctx, service, metav1.UpdateOptions{})
+	verifyIngress(ctx, t, env)
+}
+
+func TestDeployAllInOnePostgresKuma(t *testing.T) {
+	t.Log("configuring all-in-one-postgres.yaml manifest test")
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	addons := []clusters.Addon{}
+	addons = append(addons, metallb.New())
+	addons = append(addons, kuma.New())
+	if b, err := loadimage.NewBuilder().WithImage(imageLoad); err == nil {
+		addons = append(addons, b.Build())
+	}
+	builder := environments.NewBuilder().WithAddons(addons...)
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.ParseTolerant(clusterVersionStr)
+		require.NoError(t, err)
+		builder.WithKubernetesVersion(clusterVersion)
+	}
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, env.Cleanup(ctx))
+	}()
+
+	t.Log("deploying kong components")
+	manifest, err := getTestManifest(t, postgresPath)
+	require.NoError(t, err)
+	deployment := deployKong(ctx, t, env, manifest)
+
+	t.Log("this deployment used a postgres backend, verifying that postgres migrations ran properly")
+	verifyPostgres(ctx, t, env)
+
+	t.Log("adding Kuma mesh")
+	kuma.EnableMeshForNamespace(ctx, env.Cluster(), "kong")
+	kuma.EnableMeshForNamespace(ctx, env.Cluster(), "default")
+
+	// scale to force a restart of pods and trigger mesh injection (we can't annotate the Kong namespace in advance,
+	// it gets clobbered by deployKong()). is there a "rollout restart" in client-go? who knows!
+	scale := &autoscalingv1.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deployment.Name,
+			Namespace: deployment.Namespace,
+		},
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: 0,
+		},
+	}
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).UpdateScale(ctx,
+		deployment.Name, scale, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	scale.Spec.Replicas = 2
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).UpdateScale(ctx,
+		deployment.Name, scale, metav1.UpdateOptions{})
+	require.NoError(t, err)
```

Comment:
We can wrap this up in a separate func (and copy this comment there). At least it will be more obvious what (this block of code) does, it will have a name and will help with DRY

Reply:
Left this as-is since we only have two and they're identical--my rule of thumb for DRY is having 3+ instances, usually ones that vary in some way. This doesn't feel long enough that I'd want to break it out otherwise.